app/components/provider_interface/offer_summary_list_component.rb

```diff
@@ -16,17 +18,68 @@ def rows
         },
         {
           key: 'Provider',
-          value: application_choice.offered_course.provider.name,
+          value: course_option.course.provider.name,
+          change_path: change_path(:provider), action: 'training provider'
         },
         {
           key: 'Course',
-          value: application_choice.offered_course.name_and_code,
+          value: course_option.course.name_and_code,
+          change_path: change_path(:course), action: 'course'
         },
         {
           key: 'Location',
-          value: application_choice.offered_site.name_and_address,
+          value: course_option.site.name_and_address,
+          change_path: change_path(:course_option), action: 'location'
         },
       ]
     end
+
+    def paths
+      Rails.application.routes.url_helpers
+    end
+
+    # paths produced preserve the current (unsaved) provider/course/option selection
+    def change_path(target)
```

Comment:
Were we planning to push these up into the controller?

Reply:
Pushed